### ISSUE TITLE
false positive(?): flags proto fields which are being written, not read

### ISSUE THREAD

**Issue by spencerschrock** [ghostiam/protogetter#4]
I have some code [here](
https://github.com/ossf/scorecard/blob/0e3a5233ae322fbcaad589da574c9add8c4b1279/cron/internal/controller/main.go#L154-L163) which sets fields in a proto.
```go
	metadata := data.ShardMetadata{
		NumShard:  new(int32),
		ShardLoc:  new(string),
		CommitSha: new(string),
	}
	*metadata.NumShard = shardNum + 1
	*metadata.ShardLoc = bucket + "/" + data.GetBlobFilename("", t)
	*metadata.CommitSha = version.GetVersionInfo().GitCommit
	metadataJSON, err := protojson.Marshal(&metadata)
```

The linter flags this:
```
cron/internal/controller/main.go:159:3: avoid direct access to proto field metadata.NumShard, use metadata.GetNumShard() instead (protogetter)
        *metadata.NumShard = shardNum + 1
```

The provided fix (when I use the `--fix` flag) doesn't compile.
```go
	*metadata.GetNumShard() = shardNum + 1
	*metadata.GetShardLoc() = bucket + "/" + data.GetBlobFilename("", t)
	*metadata.GetCommitSha() = version.GetVersionInfo().GitCommit
```

```
invalid operation: cannot indirect metadata.GetShardLoc() (value of type string)compiler (InvalidIndirection)
```

Should the linter be flagging non-get operations?

**Comment by ghostiam** [comment]
Thanks for the issue, yes, you are right, this is a false positive.
I'll fix it as soon as I can.

**Comment by ghostiam** [comment]
Fixed in version v0.2.4
Check, please.
Thanks for the issue!
